source/agora/consensus/ValidatorSet.d

```diff
@@ -436,86 +406,88 @@ public class ValidatorSet
 
     /***************************************************************************
 
-        Add a pre-image information to a validator data
+        Get the previously stored preimage and distance for the provided
+        enrollment key, if such preimage exists and the provided distance
+        is newer.
 
         Params:
-            preimage = the pre-image information to add
+            enroll_hash = key to look up
+            new_distance = must be greater than the stored distance,
+                           otherwise false is returned
+            preimage = will contain the preimage if found
+            distance = will contain the distance if found
 
         Returns:
-            true if the pre-image information has been added to the validator
+            Return true if the preimage was found
 
     ***************************************************************************/
 
-    public bool addPreimage (const ref PreImageInfo preimage) @safe nothrow
+    private bool getOlderPreimageDistance (Hash enroll_hash, ushort new_distance,
+        out Hash preimage, out ushort distance) @trusted nothrow
```

Comment:
You can merge this with `getPreimage` if you change the signature of `getPreimage` to:
```D
public PreImageInfo getPreimage (const ref Hash enroll_key, ushort min_distance = 0) @trusted nothrow
```

Reply:
Haha I made the same comment just a minute before you. :)